include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp

```diff
@@ -538,9 +540,18 @@ class ParticleCalorimeter : public ISimulationPlugin
         DataConnector &dc = Environment<>::get().DataConnector();
         auto particles = dc.get< ParticlesType >( speciesName, true );
 
-        PMACC_KERNEL(KernelParticleCalorimeter{})
-                (grid, mapper.getSuperCellSize())
-                (particles->getDeviceParticlesBox(), (MyCalorimeterFunctor)*this->calorimeterFunctor, mapper);
+        constexpr uint32_t numWorkers = pmacc::traits::GetNumWorkers<
+            pmacc::math::CT::volume< SuperCellSize >::type::value
+        >::value;
+
+        PMACC_KERNEL( KernelParticleCalorimeter< numWorkers >{ } )(
+            grid,
+            numWorkers
+        )(
+            particles->getDeviceParticlesBox( ),
+            (MyCalorimeterFunctor)*this->calorimeterFunctor,
```

Comment:
here it is casted, above it is not

Reply:
Because this is a shared pointer (boost::shared_ptr), above it is the correct type.

Reply:
I meant this line vs. line 469 and the cast of `this->calorimeterFunctor`

Reply:
The cast can be removed the deference is casting to the correct type.